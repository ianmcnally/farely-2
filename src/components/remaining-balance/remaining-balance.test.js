import React from 'react'
import renderShallow from 'render-shallow'
import RemainingBalance from './remaining-balance'

describe('with required props', () => {
  let component

  beforeAll(() => {
    component = renderShallow(<RemainingBalance />).output
  })

  it('renders the input', () => {
    expect(component).toMatchSnapshot()
  })

})

describe('with props.onChange', () => {
  const props = { onChange(){} }
  let component

  beforeAll(() => {
    component = renderShallow(<RemainingBalance {...props} />).output
  })

  it('renders the input with an onChange handler', () => {
    expect(component).toMatchSnapshot()
  })

})

describe('with props.value', () => {
  const props = { value: 'hey' }
  let component

  beforeAll(() => {
    component = renderShallow(<RemainingBalance {...props} />).output
  })

  it('renders the input with a value', () => {
    expect(component).toMatchSnapshot()
  })

})
