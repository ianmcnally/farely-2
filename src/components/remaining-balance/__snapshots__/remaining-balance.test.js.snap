exports[`with props.onChange renders the input with an onChange handler 1`] = `
<p>
  <label
    className="db fw6 lh-copy f5"
    htmlFor="remaining-balance">
    Current balance
  </label>
  <CurrencyInput
    autoFocus={true}
    className="pa2 w-100 input-reset ba measure b--black-20"
    id="remaining-balance"
    onChange={[Function]}
    type="number"
    value={undefined} />
</p>
`;

exports[`with props.value renders the input with a value 1`] = `
<p>
  <label
    className="db fw6 lh-copy f5"
    htmlFor="remaining-balance">
    Current balance
  </label>
  <CurrencyInput
    autoFocus={true}
    className="pa2 w-100 input-reset ba measure b--black-20"
    id="remaining-balance"
    onChange={undefined}
    type="number"
    value="hey" />
</p>
`;

exports[`with required props renders the input 1`] = `
<p>
  <label
    className="db fw6 lh-copy f5"
    htmlFor="remaining-balance">
    Current balance
  </label>
  <CurrencyInput
    autoFocus={true}
    className="pa2 w-100 input-reset ba measure b--black-20"
    id="remaining-balance"
    onChange={undefined}
    type="number"
    value={undefined} />
</p>
`;
